src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,243 @@
+  private static Mechanism2d mech = new Mechanism2d(2, 2);
+  private static ComplexWidget widg;
+  private DigitalInput elvLimitSwitch;
+  private MechanismLigament2d elevatorMech;
+  private MotionMagicConfigs elvMotionMagic =
+      new MotionMagicConfigs()
+          .withMotionMagicCruiseVelocity(MMVel)
+          .withMotionMagicAcceleration(MMAcc)
+          .withMotionMagicJerk(MMJerk);
+
+  public static final SoftwareLimitSwitchConfigs elvSoftLimit =
+      new SoftwareLimitSwitchConfigs()
+          .withForwardSoftLimitEnable(true)
+          .withForwardSoftLimitThreshold(heightToRotations(Constants.ElevatorConstants.kTopLimit))
+          .withReverseSoftLimitEnable(true)
+          .withReverseSoftLimitThreshold(
+              heightToRotations(Constants.ElevatorConstants.kBottomLimit));
+  public static final Slot0Configs loopCfg =
+      new Slot0Configs()
+          .withKP(Constants.ElevatorConstants.P)
+          .withKI(Constants.ElevatorConstants.I)
+          .withKD(Constants.ElevatorConstants.D);
+  private ElevatorSim m_elevatorSim;
+
+  public Elevator() {
+    // TODO: set up for canivore
+    currentLimit.StatorCurrentLimit = 90;
+    currentLimit.SupplyCurrentLimit = 70;
+    currentLimit.SupplyCurrentLowerLimit = 40;
+    currentLimit.SupplyCurrentLowerTime = 1;
+    currentLimit.StatorCurrentLimitEnable = true;
+    currentLimit.SupplyCurrentLimitEnable = true;
+    elevatorMotor1 = new TalonFX(MotorIDConstants.kElevatorMotor1, Constants.RIOName);
+    elevatorMotor2 = new TalonFX(MotorIDConstants.kElevatorMotor2, Constants.RIOName);
+    elevatorMotor1.getConfigurator().apply(loopCfg);
+    elevatorMotor1.getConfigurator().apply(elvSoftLimit);
+    elevatorMotor1.getConfigurator().apply(currentLimit);
+    elevatorMotor1.getConfigurator().apply(invertMotor);
+    elevatorMotor1.getConfigurator().apply(elvMotionMagic);
+    elevatorMotor2.setControl(new Follower(MotorIDConstants.kElevatorMotor1, true));
+    elvLimitSwitch = new DigitalInput(Constants.ElevatorConstants.elvLimitID);
+    new Trigger(elvLimitSwitch::get).onTrue(zeroMotorEncoder());
+
+    // Simulation
+    if (Robot.isSimulation()) {
+      simElvMotor1 = elevatorMotor1.getSimState();
+      simElvMotor1.Orientation = ChassisReference.Clockwise_Positive;
+      m_elevatorSim =
+          new ElevatorSim(
+              kElevatorGearbox,
+              kElevatorGearing,
+              kCarriageMass.in(Kilograms),
+              kElevatorDrumRadius.in(Meters),
+              kMinElevatorHeight.in(Meters),
+              kMaxElevatorHeight.in(Meters),
+              true,
+              0);
+      elevatorMech =
+          mech.getRoot("root", 1, 0)
+              .append(
+                  new MechanismLigament2d(
+                      "Elevator Mech [0]", 1, 90, 10, new Color8Bit(Color.kPurple)));
+
+      if (widg == null) {
+        widg = Shuffleboard.getTab(getName()).add("Elevator", mech);
```

Comment:
Can't you use Advantagekit to log mech2ds? I could be wrong, but that would be a cleaner way of doing this I think, if it works.